mart/utils/adapters.py

```diff
@@ -36,3 +38,23 @@ def __call__(self, *args, **kwargs):
         assert callable(function)
 
         return function(*args, **kwargs)
+
+
+class PartialInstanceWrapper:
+    """Make a partial class object callable."""
+
+    def __init__(self, partial: Callable, redirecting_fn: str):
+        """
+
+        Args:
+            partial (Callable): A partial of a class object.
+            redirecting_fn (str): name of the function that will be invoked in the `__call__` method.
+        """
+        self.partial = partial
+        self.redirecting_fn = redirecting_fn
+
+    def __call__(self, *args: Any, **kwargs: Any) -> Callable:
+        # Turn a partial to a class object.
+        instance = self.partial(*args, **kwargs)
+        # Make the object callable.
+        return CallableAdapter(instance, self.redirecting_fn)
```

Comment:
This just copies the functionality of CallableAdapter. Can you rewrite this so that you don't tightly couple `PartialInstanceWrapper` with `CallableAdapter`?

Reply:
Fixed.

Reply:
Looks good!